spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+    {
+        if (numBytes < 0)
+            _throwIAE();
+
+        _generateSeed(result, numBytes);
+    }
+
+
+    fun *.getAlgorithm (@target self: SecureRandom): String
+    {
+        result = this.algorithm;
+    }
+
+
+    @final fun *.getProvider (@target self: SecureRandom): Provider
+    {
+        result = this.provider;
+    }
+
+
+    // within java.util.Random
+    fun *.ints (@target self: SecureRandom): IntStream
+    {
+        val mass: array<int> = _generateRandomIntegerArray(MAX_RANDOM_STREAM_SIZE);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new IntStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.ints (@target self: SecureRandom, randomNumberOrigin: int, randomNumberBound: int): IntStream
+    {
+        val mass: array<int> = _generateRandomIntegerArrayWithBounds(MAX_RANDOM_STREAM_SIZE, randomNumberOrigin, randomNumberBound);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new IntStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.ints (@target self: SecureRandom, streamSize: long): IntStream
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<int> = _generateRandomIntegerArray(size);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new IntStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.ints (@target self: SecureRandom, streamSize: long, randomNumberOrigin: int, randomNumberBound: int): IntStream
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<int> = _generateRandomIntegerArrayWithBounds(size, randomNumberOrigin, randomNumberBound);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new IntStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.longs (@target self: SecureRandom): LongStream
+    {
+        val mass: array<long> = _generateRandomLongArray(MAX_RANDOM_STREAM_SIZE);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new LongStreamAutomaton(state = Initialized,
+            storage = mass,
```

Comment:
Why not call `_generateRandomLongArray` directly here?

Reply:
Fixed;